### DISCUSSION TITLE
Annotated text - use tooltips?

### DISCUSSION THREAD

**Discussion by weitzman** [squidfunk/mkdocs-material#4923] [Q&A] [answered]
Hi all. I'm hoping to add an annotation presentation similar to https://genius.com/Thomas-jefferson-the-declaration-of-independence-annotated. The key features are that small strings from the text are highlighted and when hovered (or clicked, on mobile), a rich popup is shown. 

1. Is this best done on mkdocs-material with the Tooltip feature? or maybe something like the Search modal? I like how there is scrolling built into the modal.
2. Also, any tips on CSS that will make a cool looking popup like on that genius.com page?

As an aside, I would consider using [sidenotes](https://github.com/squidfunk/mkdocs-material/discussions/4688) if they were available.

**Comment by squidfunk** [comment, marked as answer]
This is kind of what we're planning to do with [instant previews](https://twitter.com/squidfunk/status/1466794654213492743). It's currently not possible, because [tooltips](https://squidfunk.github.io/mkdocs-material/reference/tooltips/) can only include pure text values – no markup. However, we're planning to add glossary-like functionality, so that users can attach arbitrary markup to almost anything. It's planned for this year (wanted to do it last year, but there was so much other stuff to do ;-))

In the meantime, you can use [annotations](https://squidfunk.github.io/mkdocs-material/reference/annotations/), with which you can achieve almost what you want!

**Comment by weitzman** [comment]
I initially ruled out annotations because I thought the plus icon was too subtle a marker to draw the users attention. But now I see that I can highlight all the annotated text by customizing the `.annotate` CSS class. Thats great. Any guidance on how I can trigger the display of the annotation when any text inside of `.annotate` is hovered/touched? I would make sure to only have 1 annotation inside of each `.annotate` block. This is how https://genius.com/Thomas-jefferson-the-declaration-of-independence-annotated works.

Thanks so much.

**Reply by squidfunk** [reply]
You might be able to somehow simulate/trigger a click on the annotation marker on hover or click of another element. I'm afraid I don't have a ready-to-use code snippet to share, but it's definitely doable with a little knowledge of JS.
